### ISSUE TITLE
Cat has a litter of five- Actually had a litter of four and an adult dead cat was labelled as the fifth

### ISSUE THREAD

**Issue by ZrainydayXXX** [ClanGenOfficial/clangen#2414]
**Type**: Code

**Describe the bug**
My cat had a litter of five and a cat who died that moon was marked as the fifth instead of it generating a kit

**Grade**: Regular

**Reproduce**
1. Have a cat bring back kits
2. On the same moon have another cat die and possibly one of the parent's kits too
3. See the error

**Commit # or Game Version Number**: 43448b39f91f8ed8e6c42d8418159d0437b116af

**Additional context**
The cat who died had high platonic like with the parent cat. Additionally, the parent cat's kit died in the same war event that moon.

**Screenshots**
![image](https://github.com/ClanGenOfficial/clangen/assets/77426212/4c194c5b-cd1a-47f7-b384-9cda5c8e3510)
![image](https://github.com/ClanGenOfficial/clangen/assets/77426212/e1f81066-a6d3-493b-a6ca-10f570d82bf6)
![image](https://github.com/ClanGenOfficial/clangen/assets/77426212/f0e7901a-89f9-4be9-9e57-33a3f35d1253)
![image](https://github.com/ClanGenOfficial/clangen/assets/77426212/a1c2fd12-cbb0-444f-80aa-e87e0b6f196e)

**Comment by ZrainydayXXX** [comment]
More context from dev discussion that I was told by Zabe to add along with the save of this very buggy clan:

- When I first made this clan as a casual story clan, I got the bug that generated extra cats. I did not report because someone else had but it's worth noting it grabbed a few cats from my other clans (Now deleted clans I believe)
- Larkkit and the dead Marigoldkit (Conkerkit in my reports) are listed as two cats' apprentices despite them being kits and one having died the moon she was born:
![image](https://github.com/ClanGenOfficial/clangen/assets/77426212/a70c0563-6d78-4b04-a650-1bc5243e1e0b)
![image](https://github.com/ClanGenOfficial/clangen/assets/77426212/539e2e6d-3005-4aff-a44a-55e175c5e3c6)
Icefur and Haystem were kits when the clan was first generated

These cats interestingly, are not listed as the kits' mentors on their pages:
![image](https://github.com/ClanGenOfficial/clangen/assets/77426212/5050b9bb-1eb2-4726-8c5f-8bbbec4dbe61)
![image](https://github.com/ClanGenOfficial/clangen/assets/77426212/bf77b1ed-9aa0-4e5e-8686-a075a0d24da2)

This clan is overall buggy so as instructed, here is the file:
[Ruined.zip](https://github.com/user-attachments/files/15746507/Ruined.zip)
[Ruinedclan.json](https://github.com/user-attachments/files/15746502/Ruinedclan.json)

**Comment by ZrainydayXXX** [comment]
Adding on, upon making a new clan, I got the above mentioned issue of extra generated cats. The ones underlined in red are from another clan. I pressed save this time but upon closing and opening again, it made no difference and they disappeared despite the fact that when I checked the code after saving, they were there. After saving again, they were gone.

![image](https://github.com/ClanGenOfficial/clangen/assets/77426212/595a1436-35e6-42b0-b536-356d9d7ca0f4)

Popstar, Breezecurrent, Privetfoot, Saffrondawn, Blackfade, Branchcloud, Parsleypaw and Darkkit were cats I chose.
Phloxecho, Bonegrowl, Icehaze, Ottercloud, Cinderkit, Thistlekit, Brightkit and Charredkit are the bugged cats.
